solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -177,27 +180,31 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         return amount;
     }
 
-    function _write() internal {
-        uint256 price0 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                1
-            );
-
-        uint256 price1 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                1
-            );
-
-        oracle.write(price0, price1);
-    }
+	function _write() internal {
+		uint256 price0 = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
+			reserveBalance(reserveTokens[0]),
+			reserves[reserveTokens[0]].weight,
+			reserveBalance(reserveTokens[1]),
+			reserves[reserveTokens[1]].weight,
+			reserveBalance(reserveTokens[0]).div(POOL_PERCENTAGE)

```

Comment:
you set the percentage to 10. if you really mean "10% of the pool should be swapped" with it, this code here would only work because you picked 10 and not for any other number. if i would set it to 20% for example, you would be deviding by 20 here and therefore calculating 5%. if it is intended to be used this way, you should rename the variable

Reply:
Yes